components/dashboard/src/settings/Preferences.tsx

```diff
@@ -16,73 +15,60 @@ import { getGitpodService } from "../service/service";
 import { ThemeContext } from "../theme-context";
 import { UserContext } from "../user-context";
 import settingsMenu from "./settings-menu";
+import IDENone from '../icons/IDENone.svg';
+import IDENoneDark from '../icons/IDENoneDark.svg';
 
 type Theme = 'light' | 'dark' | 'system';
 
+const DesktopNoneId = "none";
+const DesktopNone: IDEOption = {
+    "image": "",
+    "logo": IDENone,
+    "orderKey": "-1",
+    "title": "None",
+    "type": "desktop"
+};
+
 export default function Preferences() {
     const { user } = useContext(UserContext);
-    const { setIsDark } = useContext(ThemeContext);
-
-    const [defaultIde, setDefaultIde] = useState<string>(user?.additionalData?.ideSettings?.defaultIde || "");
-    const actuallySetDefaultIde = async (value: string) => {
-        const additionalData = user?.additionalData || {};
-        const settings = additionalData.ideSettings || {};
-        settings.defaultIde = value;
+    const { setIsDark, isDark } = useContext(ThemeContext);
+
+    const updateUserIDEInfo = async (defaultDesktopIde: string, defaultIde: string) => {
+        const useDesktopIde = defaultDesktopIde !== DesktopNoneId;
+        const additionalData = user?.additionalData ?? {};
+        const settings = additionalData.ideSettings ?? {};
+        settings.useDesktopIde = useDesktopIde;
+        settings.defaultIde = defaultIde;
+        settings.defaultDesktopIde = defaultDesktopIde;
```

Comment:
**Thought**: Here, you might be setting `"none"` as the `defaultDesktopIde` in `user.additionalData.ideSettings`.

It seems a little strange that `user.additionalData.ideSettings.defaultDesktopIde` can have a value that is not present in [the configmap](https://github.com/gitpod-io/gitpod/blob/4bdcf8c8f203795c2951a19940e3151a57795f1e/chart/templates/server-ide-configmap.yaml#L61-L103).

Maybe, just to be safe, we should add the `none` option in the configmap, or set `defaultDesktopIde` to `undefined` instead of `"none"`?